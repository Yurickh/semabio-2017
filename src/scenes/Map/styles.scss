@import 'ambience/color';
@import 'ambience/responsive';
@import 'base/responsive-prop';

.page-map {
  width: 100%;

  @include responsive-prop(margin-top, 100, 170);
  @include small-only {
    width: 100%;
  }

  & > .simple-title > .divisor {
    @include responsive-prop(margin-bottom, 24, 24);
  }

  & > .subtitle {
    width: 82vw;
    margin: 0 auto;

    text-align: center;
    letter-spacing: .41px;

    opacity: .8;
    color: $color-description;

    @include responsive-prop(font-size, 20, 26);
    @include responsive-prop(line-height, 26, 32);
  }

  & > .content {
    display: flex;

    @include responsive-prop(margin-top, 42, 42);
    @include small-only {
      flex-direction: column;
    }
  }

  & > .content > .map-container {
    width: 50vw;

    background-color: red;

    @include responsive-prop(height, 450, 450);
    @include small-only {
      width: 100vw;
    }
  }

  & > .content > .bus-lines {
    color: $color-bullfrog;

    @include responsive-prop(font-size, 20, 26);
    @include responsive-prop(margin-top, 23, 23);
    @include responsive-prop(margin-left, 54, 54);
    @include small-only {
      width: 82vw;
      margin: 0 auto;

      @include responsive-prop(margin-top, 23, 23);
    }
  }

  & > .content > .bus-lines > .line {
    @include responsive-prop(margin-top, 27, 27);
  }

  & > .content > .bus-lines > .simple-button {
    @include responsive-prop(margin-top, 45, 45);
    @include responsive-prop(width, 266, 266);
    @include responsive-prop(padding, 17, 17);
    @include small-only {
      width: 82vw;
      margin: 0 auto;

      @include responsive-prop(margin-top, 45, 45);
    }
  }
}
