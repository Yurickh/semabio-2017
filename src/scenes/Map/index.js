import React, { Component } from 'react'

import Title from '../../components/Title'
import Button from '../../components/Button'
import GoogleMap from '../../components/GoogleMap'

import './styles.css'

class Map extends Component {
  render() {
    return (
      <section className="page-map">
        <Title>Como chegar</Title>

        <h2 className="subtitle">Selecionamos algumas das principais linhas de ônibus para você</h2>

        <div className="content">
          <main className="map-container">
            <GoogleMap
              lat={-15.7630982}
              lng={-47.872239}
              zoom={14}
            />
          </main>

          <aside className="bus-lines">
            <div className="line">110 - Rod. do Plano Piloto</div>
            <div className="line">167 - Guará</div>
            <div className="line">349 - Taguatinga</div>
            <div className="line">521 - Sobradinho</div>
            <div className="line">605 - Planaltina</div>
            <div className="line">160.2 - Núcleo Bandeirante</div>
            <a
              target="_blank"
              rel="noopener noreferrer"
              href="http://www.sistemas.dftrans.df.gov.br/horarios/"
            >
              <Button color="green">Mais Linhas</Button>
            </a>
            <h3 className="observation">O evento vai rolar na Universidade de Brasília - Instituto de Ciências Biológicas</h3>
          </aside>
        </div>
      </section>
    )
  }
}

export default Map
