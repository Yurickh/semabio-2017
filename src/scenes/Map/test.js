import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Map from '../Map'

describe('Map', () => {
  it('should render correctly', () => {
    const page = shallow(<Map />)
    expect(toJSON(page)).toMatchSnapshot()
  })
})
