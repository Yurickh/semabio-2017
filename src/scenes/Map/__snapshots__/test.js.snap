// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should render correctly 1`] = `
<section
  className="page-map"
>
  <Title>
    Como chegar
  </Title>
  <h2
    className="subtitle"
  >
    Selecionamos algumas das principais linhas de ônibus para você
  </h2>
  <div
    className="content"
  >
    <main
      className="map-container"
    >
      <GoogleMap
        lat={-15.7630982}
        lng={-47.872239}
        zoom={14}
      />
    </main>
    <aside
      className="bus-lines"
    >
      <div
        className="line"
      >
        167.2 - Núcleo Bandeirante
      </div>
      <div
        className="line"
      >
        521 - Sobradinho
      </div>
      <div
        className="line"
      >
        517 - Sobradinho II
      </div>
      <div
        className="line"
      >
        167.2 - Núcleo Bandeirante
      </div>
      <div
        className="line"
      >
        521 - Sobradinho
      </div>
      <div
        className="line"
      >
        517 - Núcleo Bandeirante
      </div>
      <a
        href="http://www.sistemas.dftrans.df.gov.br/horarios/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Button
          color="green"
        >
          Mais Linhas
        </Button>
      </a>
    </aside>
  </div>
</section>
`;
